python/pyiceberg/avro/reader.py

```diff
@@ -238,26 +249,25 @@ def skip(self, decoder: BinaryDecoder) -> None:
             return self.option.skip(decoder)
 
 
-class StructProtocolReader(Reader):
-    create_struct: Callable[[], StructProtocol]
+class StructReader(Reader):
     fields: Tuple[Tuple[Optional[int], Reader], ...]
+    create_struct: Type[StructProtocol]
 
-    def __init__(self, fields: Tuple[Tuple[Optional[int], Reader], ...], create_struct: Callable[[], StructProtocol]):
-        self.create_struct = create_struct
+    def __init__(self, fields: Tuple[Tuple[Optional[int], Reader], ...], create_struct: Type[StructProtocol] = Record):
         self.fields = fields
-
-    def create_or_reuse(self, reuse: Optional[StructProtocol]) -> StructProtocol:
```

Comment:
I left out the reuse for now. This conflicted with the default values. For example, the DataFile has `DataFileContent.DATA` as the default value for `content`. If we want to re-use we also need to re-set the defaults (this is quite easy since we can just loop over the Pydantic fields).

Reply:
Makes sense. I think we can do it later. We'd want to include the defaults in the type system and replace the `NoneReader` with `DefaultReader(defaultValue)`